playwright/containers/n8n-test-container-dependencies.ts

```diff
@@ -0,0 +1,166 @@
+import { PostgreSqlContainer } from '@testcontainers/postgresql';
+import { RedisContainer } from '@testcontainers/redis';
+import type { StartedNetwork, StartedTestContainer } from 'testcontainers';
+import { GenericContainer, Wait } from 'testcontainers';
+
+export async function setupRedis(
+	redisImage: string,
+	uniqueSuffix: string,
+	network: StartedNetwork,
+): Promise<StartedTestContainer> {
+	return await new RedisContainer(redisImage)
+		.withNetwork(network)
+		.withNetworkAliases('redis')
+		.withLabels({
+			'com.docker.compose.project': uniqueSuffix,
+			'com.docker.compose.service': 'redis',
+		})
+		.withName(`${uniqueSuffix}-redis`)
+		.withReuse()
+		.start();
+}
+
+export async function setupPostgres(
+	postgresImage: string,
+	uniqueSuffix: string,
+	network: StartedNetwork,
+): Promise<{
+	container: StartedTestContainer;
+	database: string;
+	username: string;
+	password: string;
+}> {
+	const postgres = await new PostgreSqlContainer(postgresImage)
+		.withNetwork(network)
+		.withNetworkAliases('postgres')
+		.withDatabase('n8n_db')
+		.withUsername('n8n_user')
+		.withPassword('test_password')
+		.withStartupTimeout(30000)
+		.withLabels({
+			'com.docker.compose.project': uniqueSuffix,
+			'com.docker.compose.service': 'postgres',
+		})
+		.withName(`${uniqueSuffix}-postgres`)
+		.withReuse()
+		.start();
+
+	return {
+		container: postgres,
+		database: postgres.getDatabase(),
+		username: postgres.getUsername(),
+		password: postgres.getPassword(),
+	};
+}
+
+/**
+ * Setup NGINX for multi-main instances
+ * @param nginxImage The Docker image for NGINX.
+ * @param uniqueSuffix A unique suffix for naming and labeling.
+ * @param mainInstances An array of running backend container instances.
+ * @param network The shared Docker network.
+ * @param nginxPort The host port to expose for NGINX.
+ * @returns A promise that resolves to the started NGINX container.
+ */
+export async function setupNginxLoadBalancer(
+	nginxImage: string,
+	uniqueSuffix: string,
+	mainInstances: StartedTestContainer[],
+	network: StartedNetwork,
+): Promise<StartedTestContainer> {
+	// Generate upstream server entries from the list of main instances.
+	const upstreamServers = mainInstances
+		.map((_, index) => `  server ${uniqueSuffix}-n8n-main-${index + 1}:5678;`)
+		.join('\n');
+
+	// Define the NGINX configuration with dynamic upstream servers.
+	const nginxConfig = `
```

Comment:
We could extract this into a separate function (eg `buildNginxConfig`) to make this method a bit shorted and readable